src/Meshes/subMesh.ts

```diff
@@ -404,12 +404,29 @@ export class SubMesh extends BaseSubMesh implements ICullable {
     private _intersectTriangles(ray: Ray, positions: Vector3[], indices: IndicesArray,
         fastCheck?: boolean, trianglePredicate?: TrianglePickingPredicate): Nullable<IntersectionInfo> {
         var intersectInfo: Nullable<IntersectionInfo> = null;
-        // Triangles test
-        for (var index = this.indexStart; index < this.indexStart + this.indexCount; index += 3) {
-            var p0 = positions[indices[index]];
-            var p1 = positions[indices[index + 1]];
-            var p2 = positions[indices[index + 2]];
 
+        var start = this.indexStart;
+        var count = this.indexCount;
+
+        if(!indices.length) {
+            start = this.verticesStart;
+            count = this.verticesCount;
+        }
+
+        // Triangles test
+        var p0, p1, p2;
+        for (var index = start; index < start + count; index += 3) {
+            if(!indices.length) {

```

Comment:
I would recommend a second function for perf reason. Picking can go through complicated meshes and avoid one if per face can be great

Reply:
ok, lint seemed like it was just unhappy with my missing whitespaces, hope i got them all, will check once it's done running, then update to a separate function

Reply:
sounds good!